app/src/main/java/com/keylesspalace/tusky/TabData.kt

```diff
@@ -43,63 +43,81 @@ data class TabData(
     val fragment: (List<String>) -> Fragment,
     val arguments: List<String> = emptyList(),
     val title: (Context) -> String = { context -> context.getString(text) }
-)
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as TabData
+
+        if (id != other.id) return false
+        if (arguments != other.arguments) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
```

Comment:
Is it used as hash key somewhere?

Reply:
Doesn't matter. If you override `equals()` you must override `hashcode()`, it's part of the contract between the two methods.